### PR TITLE
Adding proxy to update procedure

### PR DESCRIPTION
**Pull request by davidbz**

### CODE REVIEW

File: youtube_dl/YoutubeDL.py [hunk 1]
```diff
@@ -1994,6 +1994,9 @@ def get_encoding(self):
             encoding = preferredencoding()
         return encoding
 
+    def get_opener(self):
```
Comment (jaimeMF) [inline youtube_dl/YoutubeDL.py]: Directly access the opener, after all we are only using it internally.

**Comment by Kasher** [comment]
+1

File: youtube_dl/YoutubeDL.py (hunk 1 above)
Comment (davidbz) [inline youtube_dl/YoutubeDL.py]: Done

File: youtube_dl/update.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
     compat_str,
     compat_urllib_request,
```
Comment (dstftw) [inline youtube_dl/update.py]: It's unused now.

Comment (davidbz) [inline youtube_dl/update.py]: Done

**Comment by dstftw** [comment]
Thanks.

**Pull request merged**